src/charm.py

```diff
@@ -315,6 +331,9 @@ def _get_usim_opc_from_config(self) -> Optional[str]:
     def _get_usim_sequence_number_from_config(self) -> Optional[str]:
         return cast(Optional[str], self.model.config.get("usim-sequence-number"))
 
+    def _get_dnn_from_config(self) -> Optional[str]:
+        return cast(Optional[str], self.model.config.get("dnn"))
```

Comment:
Why does this return an Optional and why the cast? The config is already of type string and has a default value. Even if the user tries to remove the default value, an empty string will evaluate as False.

Reply:
Honestly I went the same way as for all of the other configuration options:

```python
 def _get_imsi_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("imsi"))

    def _get_mcc_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("mcc"))

    def _get_mnc_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("mnc"))

    def _get_sd_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("sd"))

    def _get_sst_from_config(self) -> Optional[int]:
        return cast(Optional[int], self.model.config.get("sst"))

    def _get_tac_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("tac"))

    def _get_upf_gateway_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("upf-gateway"))

    def _get_upf_subnet_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("upf-subnet"))

    def _get_usim_key_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("usim-key"))

    def _get_usim_opc_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("usim-opc"))

    def _get_usim_sequence_number_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("usim-sequence-number"))

    def _get_dnn_from_config(self) -> Optional[str]:
        return cast(Optional[str], self.model.config.get("dnn"))
```